pkg/controller/constraint/constraint_controller.go

```diff
@@ -151,12 +188,55 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 			if err := r.Update(context.Background(), instance); err != nil {
 				return reconcile.Result{Requeue: true}, nil
 			}
+			// removing constraint entry from cache
+			delete(r.constraintsCache, constraintKey)
```

Comment:
It seems like this also needs to be gated with the cache lock?

Reply:
+1 good catch

Reply:
thanks, added